cmd/remote-storage/main.go

```diff
@@ -33,93 +34,217 @@ import (
-		RunE: func(_ *cobra.Command, _ /* args */ []string) error {
-			if err := svc.Start(v); err != nil {
-				return err
-			}
-			logger := svc.Logger // shortcut
-			baseFactory := svc.MetricsFactory.Namespace(metrics.NSOptions{Name: "jaeger"})
-			metricsFactory := baseFactory.Namespace(metrics.NSOptions{Name: "remote-storage"})
-			version.NewInfoMetrics(metricsFactory)
-
-			opts, err := new(app.Options).InitFromViper(v)
-			if err != nil {
-				logger.Fatal("Failed to parse options", zap.Error(err))
-			}
+		RunE: func(cmd *cobra.Command, _ /* args */ []string) error {
+			// Check if config file is provided
+			configFile, _ := cmd.Flags().GetString(configFlagName)
 
-			baseTelset := telemetry.Settings{
-				Logger:        svc.Logger,
-				Metrics:       baseFactory,
-				ReportStatus:  telemetry.HCAdapter(svc.HC()),
-				MeterProvider: noop.NewMeterProvider(), // TODO
+			if configFile != "" {
+				// New YAML configuration path
+				return runWithYAMLConfig(cmd, v, svc, configFile)
 			}
 
-			storageFactory.InitFromViper(v, logger)
-			if err := storageFactory.Initialize(baseTelset.Metrics, baseTelset.Logger); err != nil {
-				logger.Fatal("Failed to init storage factory", zap.Error(err))
-			}
-
-			tm := tenancy.NewManager(&opts.Tenancy)
-			telset := baseTelset // copy
-			telset.Metrics = metricsFactory
-
-			traceFactory := v1adapter.NewFactory(storageFactory)
-			depFactory := traceFactory.(depstore.Factory)
-			server, err := app.NewServer(context.Background(), opts, v1adapter.NewFactory(storageFactory), depFactory, tm, telset)
-			if err != nil {
-				logger.Fatal("Failed to create server", zap.Error(err))
-			}
-
-			if err := server.Start(context.Background()); err != nil {
-				logger.Fatal("Could not start servers", zap.Error(err))
-			}
-
-			svc.RunAndThen(func() {
-				server.Close()
-				if err := storageFactory.Close(); err != nil {
-					logger.Error("Failed to close storage factory", zap.Error(err))
-				}
-			})
-			return nil
+			// Legacy CLI flags path (with deprecation warning)
+			return runWithCLIFlags(cmd, v, svc)
 		},
 	}
 
+	// Add config file flag
+	command.Flags().StringP(configFlagName, configFlagShort, "", "Path to YAML configuration file (recommended)")
+
+	// Legacy flags (deprecated)
+	v1ConfigureFlags(v, command, svc)
+
 	command.AddCommand(version.Command())
 	command.AddCommand(env.Command())
 	command.AddCommand(docs.Command(v))
 	command.AddCommand(status.Command(v, ports.RemoteStorageAdminHTTP))
 	command.AddCommand(printconfig.Command(v))
 	command.AddCommand(featuregate.Command())
 
+	if err := command.Execute(); err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+}
+
+// v1ConfigureFlags configures legacy CLI flags (deprecated).
+func v1ConfigureFlags(v *viper.Viper, command *cobra.Command, svc *flags.Service) {
+	storageFactory, err := storage.NewFactory(storage.ConfigFromEnvAndCLI(os.Args, os.Stderr))
```

Comment:
this is the part I want to remove, it's not needed and not used by jaegere_storage extension